### ISSUE TITLE
Part 5 Error: XML does not correspond to XSD schema

### ISSUE THREAD

**Issue by Enea-Carlini** [openml/openml-pytorch#27]
Hello,

I am trying to submit my model for Part 5 but I am getting the following error:

OpenMLServerException: https://www.openml.org/api/v1/xml/run/ returned code 202: Could not validate run xml by xsd - XML does not correspond to XSD schema. Error Element '{[http://openml.org/openml}setup_string](http://openml.org/openml%7Dsetup_string)': [facet 'maxLength'] The value has a length of '42121'; this exceeds the allowed maximum length of '32768'.
 on line 4 column 0. Error Element '{[http://openml.org/openml}setup_string](http://openml.org/openml%7Dsetup_string)': 'Python_3.11.3. Torch_2.6.0_cu124. NumPy_1.25.1. SciPy_1.11.1.

Or with another model:

Problem validating uploaded description file - XML does not correspond to XSD schema. Error Element '{[http://openml.org/openml}default_value](http://openml.org/openml%7Ddefault_value)': [facet 'maxLength'] The value has a length of '2862'; this exceeds the allowed maximum length of '2048'.
 on line 5641 column 0. Error Element '{http://openml.org/openml}default_value': '[{&quot;oml-python:serialized_object&quot;: &quot;component_reference&quot;, ... ALL MODEL ... &quot;denselayer24&quot;}}]' is not a valid value of the atomic type '{http://openml.org/openml}basic_latin2048'.
 on line 5641 column 0.

I have already tried re-installing the requirements, different models, both my local environment and the HPC system.
I only had success submitting the ToyModel provided.
The models I tried are below 100MB.

What should I try next? Thank you.

**Comment by viljanvr** [comment]
I also have this. After troubleshooting i found out that the problem seems to occur when you try to publish a model containing an `nn.Sequential` object with too many layers. When it serializes the object the `children` property of `nn.Sequential` becomes to long.

To recreate this you can use the [convnext small](https://pytorch.org/vision/main/models/generated/torchvision.models.convnext_small.html#torchvision.models.convnext_small) model.

Error:
```
Problem validating uploaded description file - XML does not correspond to XSD schema. Error Element '{http://openml.org/openml}default_value': [facet 'maxLength'] The value has a length of '2680'; this exceeds the allowed maximum length of '2048'.
 on line 2461 column 0..
```

Error element:
```
<oml:component>
  <oml:identifier>5</oml:identifier>
  <oml:flow xmlns:oml="http://openml.org/openml">
    <oml:name>torch.nn.Sequential.66a334137fd11b</oml:name>
    <oml:class_name>torch.nn.Sequential.66a334137fd11b</oml:class_name>
    <oml:external_version>openml==0.15.1,torch==2.1.2,torch==module.__version__,torchvision==module.__version__</oml:external_version>
    <oml:description>Automatically created pytorch flow.</oml:description>
    <oml:language>English</oml:language>
    <oml:dependencies>torch==2.1.2
numpy&gt;=1.6.1
scipy&gt;=0.9</oml:dependencies>
    <oml:parameter>
      <oml:name>0</oml:name>
      <oml:default_value>{"oml-python:serialized_object": "component_reference", "value": {"key": "0", "step_name": null}}</oml:default_value>
    </oml:parameter>
    <oml:parameter>
      <oml:name>1</oml:name>
      <oml:default_value>{"oml-python:serialized_object": "component_reference", "value": {"key": "1", "step_name": null}}</oml:default_value>
    </oml:parameter>
    <oml:parameter>
      <oml:name>10</oml:name>
      <oml:default_value>{"oml-python:serialized_object": "component_reference", "value": {"key": "10", "step_name": null}}</oml:default_value>
    </oml:parameter>
    [...]
    <oml:parameter>
      <oml:name>9</oml:name>
      <oml:default_value>{"oml-python:serialized_object": "component_reference", "value": {"key": "9", "step_name": null}}</oml:default_value>
    </oml:parameter>
    <oml:parameter>
      <oml:name>children</oml:name>
      <oml:default_value>[{"oml-python:serialized_object": "component_reference", "value": {"key": "0", "step_name": "0"}}, {"oml-python:serialized_object": "component_reference", "value": {"key": "1", "step_name": "1"}}, {"oml-python:serialized_object": "component_reference", "value": {"key": "2", "step_name": "2"}}, {"oml-python:serialized_object": "component_reference", "value": {"key": "3", "step_name": "3"}}, {"oml-python:serialized_object": "component_reference", "value": {"key": "4", "step_name": "4"}}, {"oml-python:serialized_object": "component_reference", "value": {"key": "5", "step_name": "5"}}, {"oml-python:serialized_object": "component_reference", "value": {"key": "6", "step_name": "6"}}, {"oml-python:serialized_object": "component_reference", "value": {"key": "7", "step_name": "7"}}, {"oml-python:serialized_object": "component_reference", "value": {"key": "8", "step_name": "8"}}, {"oml-python:serialized_object": "component_reference", "value": {"key": "9", "step_name": "9"}}, {"oml-python:serialized_object": "component_reference", "value": {"key": "10", "step_name": "10"}}, {"oml-python:serialized_object": "component_reference", "value": {"key": "11", "step_name": "11"}}, {"oml-python:serialized_object": "component_reference", "value": {"key": "12", "step_name": "12"}}, {"oml-python:serialized_object": "component_reference", "value": {"key": "13", "step_name": "13"}}, {"oml-python:serialized_object": "component_reference", "value": {"key": "14", "step_name": "14"}}, {"oml-python:serialized_object": "component_reference", "value": {"key": "15", "step_name": "15"}}, {"oml-python:serialized_object": "component_reference", "value": {"key": "16", "step_name": "16"}}, {"oml-python:serialized_object": "component_reference", "value": {"key": "17", "step_name": "17"}}, {"oml-python:serialized_object": "component_reference", "value": {"key": "18", "step_name": "18"}}, {"oml-python:serialized_object": "component_reference", "value": {"key": "19", "step_name": "19"}}, {"oml-python:serialized_object": "component_reference", "value": {"key": "20", "step_name": "20"}}, {"oml-python:serialized_object": "component_reference", "value": {"key": "21", "step_name": "21"}}, {"oml-python:serialized_object": "component_reference", "value": {"key": "22", "step_name": "22"}}, {"oml-python:serialized_object": "component_reference", "value": {"key": "23", "step_name": "23"}}, {"oml-python:serialized_object": "component_reference", "value": {"key": "24", "step_name": "24"}}, {"oml-python:serialized_object": "component_reference", "value": {"key": "25", "step_name": "25"}}, {"oml-python:serialized_object": "component_reference", "value": {"key": "26", "step_name": "26"}}]</oml:default_value>
    </oml:parameter>
```

**Comment by SubhadityaMukherjee** [comment]
Hey, I have patched it for now. If you reinstall using git and not pypi (as you were doing so far), it should work again. Could you try it out and let me know if it works for you now? @Enea-Carlini @viljanvr

**Comment by Enea-Carlini** [comment]
Hello, I tried to do:

pip uninstall openml-pytorch
pip install git+https://github.com/openml/openml-pytorch.git

But I still get the same error:

OpenMLServerException: https://www.openml.org/api/v1/xml/run/ returned code 202: Could not validate run xml by xsd - XML does not correspond to XSD schema. Error Element '{[http://openml.org/openml}setup_string](http://openml.org/openml%7Dsetup_string)': [facet 'maxLength'] The value has a length of '42120'; this exceeds the allowed maximum length of '32768'.
 on line 4 column 0. Error Element '{[http://openml.org/openml}setup_string](http://openml.org/openml%7Dsetup_string)': 'Python_3.11.9. Torch_2.6.0_cu126. NumPy_2.2.0. SciPy_1.15.2.

Did I do something wrong? @SubhadityaMukherjee

**Comment by Lunashieh** [comment]
Same problem, also ‘pip uninstall openml-pytorch, pip install git+https://github.com/openml/openml-pytorch.git’ doesn‘t work for me.

**Comment by viljanvr** [comment]
I also updated using `pip uninstall openml-pytorch && pip install git+https://github.com/openml/openml-pytorch` and I still get the same error.

The serialised XML code is the exact same as before the update.

**Comment by SubhadityaMukherjee** [comment]
@viljanvr could you email me your code? My email is in the forums, dont want to post it publicly here :)

**Comment by SubhadityaMukherjee** [comment]
Ah I think I found the problem. Give me a few minutes, I am working on a fix for it.

**Comment by SubhadityaMukherjee** [comment]
Alright maybe not, this might take some more time to test properly. Please check back tomorrow ):

**Comment by SubhadityaMukherjee** [comment]
Took some more time but for future reference, this related to increasing the limit for the XML default type.
It should be fine now

**Comment by Enea-Carlini** [comment]
Hello @SubhadityaMukherjee,

I uninstalled and reinstalled as indicated previously, but I am still unable to send any model apart from the ToyModel.
The error is still:
OpenMLServerException: https://www.openml.org/api/v1/xml/run/ returned code 202: Could not validate run xml by xsd - XML does not correspond to XSD schema. Error Element '{[http://openml.org/openml}setup_string](http://openml.org/openml%7Dsetup_string)': [facet 'maxLength'] The value has a length of '42120'; this exceeds the allowed maximum length of '32768'.
 on line 4 column 0. Error Element '{[http://openml.org/openml}setup_string](http://openml.org/openml%7Dsetup_string)': 'Python_3.11.9. Torch_2.6.0_cu126. NumPy_2.2.0. SciPy_1.15.2. ... ANY MODEL ...' is not a valid value of the atomic type '{[http://openml.org/openml}basic_latin32768](http://openml.org/openml%7Dbasic_latin32768)'.
 on line 4 column 0.

What should I do? Could you please share any suggestions? Is there a subset of models that work fine?

**Comment by Enea-Carlini** [comment]
I don't know if it's related, but on OpenML.org, if I log in and check the runs in my account, despite showing "1 run", when I click on that no run appears. I would guess that single run is indeed the ToyModel that also makes me visible on the leaderboard.
Not sure if this can help @SubhadityaMukherjee

**Comment by Enea-Carlini** [comment]
Hello @SubhadityaMukherjee,

I am still getting the same error, sometimes even with models with an XML of length 2862.
I am not sure why the ticket has been closed.
Apparently some models have been uploaded either way, because on my profile there are 6 runs, but I cannot see the details of these runs to be able to know which model achieved which accuracy, how can I solve this?
